pkgs/by-name/zl/zluda/package.nix

```diff
@@ -0,0 +1,79 @@
+{ lib, fetchFromGitHub, rocmPackages, python3, cargo, rustc, cmake, clang, zlib, libxml2, libedit, rustPlatform, stdenv }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "zluda";
+  version = "3";
+
+  src = fetchFromGitHub {
+    owner = "vosen";
+    repo = "ZLUDA";
+    rev = "v${version}";
+    hash = "sha256-lykM18Ml1eeLMj/y6uPk34QOeh7Y59i1Y0Nr118Manw=";
+    fetchSubmodules = true;
+  };
+
+  cargoPatches = [ ./ZLUDA-cargo.lock.patch ];
+
+  buildInputs = [
+    rocmPackages.clr
+    rocmPackages.miopen
+    rocmPackages.rocm-smi
+    rocmPackages.rocsparse
+    rocmPackages.rocsolver
+    rocmPackages.rocblas
+    rocmPackages.hipblas
+    rocmPackages.rocm-cmake
+    rocmPackages.hipfft
+  ];
+
+  nativeBuildInputs = [
+    python3
+    cargo
+    rustc
+    cmake
+    clang
+    zlib
+    libxml2
+    libedit
+  ];
+
+  preConfigure = ''
+    # Comment out zluda_blaslt in Cargo.toml
+    sed -i '/zluda_blaslt/d' Cargo.toml
+    # TODO: investigate test failure (the test seems to require build time env vars that aren't set on linux?)
+    rm zluda_inject/tests/inject.rs
```

Comment:
A reference to a more detailed documentation of the errors is desirable. A gist with the build logs or, better yet, a github issue. A very short textual description of the issue so people can't get the idea without opening the link

Reply:
* For inject: `disable test written for windows only: https://github.com/vosen/ZLUDA/blob/774f4bcb37c39f876caf80ae0d39420fa4bc1c8b/zluda_inject/tests/inject.rs#L55`? (Though I see quite a few packages that just set `doCheck = false;` with no explanation or comments like "tests fail".
* For blaslt: Sorry, already forgot, you'll have to reinvestigate this @errnoh

Reply:
Added the above note for the `inject.rs`, starting a separate conversation to address the `zluda_blaslt` part